src/beaconer.rs

```diff
@@ -113,15 +125,33 @@ impl Beaconer {
         };
         let beacon_id = beacon.beacon_id();
         info!(logger, "transmitting beacon"; "beacon" => &beacon_id);
-        let report = match poc_lora::LoraBeaconReportReqV1::try_from(beacon.clone()) {
+        let mut report = match poc_lora::LoraBeaconReportReqV1::try_from(beacon.clone()) {
             Ok(report) => report,
             Err(err) => {
                 warn!(logger, "failed to construct beacon report {err:?}");
                 return;
             }
         };
-        self.transmit.transmit_beacon(beacon).await;
+        let (tx, rx) = sync::response_channel();
+        self.transmit
+            .transmit_beacon(BeaconReq {
+                beacon,
+                response: tx,
+            })
+            .await;
+        let (powe, tmst) = if let Ok(Ok(BeaconResp { powe, tmst })) = rx.recv().await {
+            info!(
+                logger,
+                "Beacon transmitted with power {powe:?} and time {tmst:?}"
+            );
+            (powe, tmst)
+        } else {
+            warn!(logger, "failed to transmit beacon");
+            return;
+        };
 
+        report.tx_power = powe.unwrap_or(27);
+        report.tmst = tmst.unwrap_or(0);
```

Comment:
If i understand correctly these should always be supplied as `Some(integer)` in the context of the beacon packets but I'd like to have a better fallback value here that doesn't feel arbitrary.